vscode/src/completions/context/context-data-logging.ts

```diff
@@ -0,0 +1,108 @@
+import type { AutocompleteContextSnippet } from '@sourcegraph/cody-shared'
+import type * as vscode from 'vscode'
+import { logDebug } from '../../log'
+import type { ContextRetriever } from '../types'
+import type { RetrievedContextResults } from './completions-context-ranker'
+import { JaccardSimilarityRetriever } from './retrievers/jaccard-similarity/jaccard-similarity-retriever'
+import { DiagnosticsRetriever } from './retrievers/recent-user-actions/diagnostics-retriever'
+import { RecentCopyRetriever } from './retrievers/recent-user-actions/recent-copy'
+import { RecentEditsRetriever } from './retrievers/recent-user-actions/recent-edits-retriever'
+import { RecentViewPortRetriever } from './retrievers/recent-user-actions/recent-view-port'
+import { RetrieverIdentifier } from './utils'
+
+interface RetrieverConfig {
+    identifier: RetrieverIdentifier
+    maxSnippets: number
+}
+
+export class ContextRetrieverDataCollection implements vscode.Disposable {
+    private readonly retrieverConfigs: ReadonlyArray<RetrieverConfig>
+    private readonly dataCollectionRetrievers: ReadonlyArray<ContextRetriever>
+    private disposables: vscode.Disposable[] = []
+    private static readonly MAX_PAYLOAD_SIZE_BYTES = 1024 * 1024 // 1 MB
+
+    constructor() {
+        this.retrieverConfigs = [
+            { identifier: RetrieverIdentifier.RecentCopyRetriever, maxSnippets: 1 },
+            { identifier: RetrieverIdentifier.RecentEditsRetriever, maxSnippets: 15 },
+            { identifier: RetrieverIdentifier.DiagnosticsRetriever, maxSnippets: 15 },
+            { identifier: RetrieverIdentifier.RecentViewPortRetriever, maxSnippets: 10 },
+            { identifier: RetrieverIdentifier.JaccardSimilarityRetriever, maxSnippets: 15 },
+        ]
```

Comment:
Since this is a constant value we can extract it from the constructor:

```suggestion
     private readonly retrieverConfigs = [
          { identifier: RetrieverIdentifier.RecentCopyRetriever, maxSnippets: 1 },
          { identifier: RetrieverIdentifier.RecentEditsRetriever, maxSnippets: 15 },
          { identifier: RetrieverIdentifier.DiagnosticsRetriever, maxSnippets: 15 },
          { identifier: RetrieverIdentifier.RecentViewPortRetriever, maxSnippets: 10 },
          { identifier: RetrieverIdentifier.JaccardSimilarityRetriever, maxSnippets: 15 },
     ]

    constructor() {
```